### PR TITLE
Automation for Release building

### PR DESCRIPTION
**Pull request by lavabyrd**
https://figmentio.atlassian.net/browse/GRAPH-337

Here's an example of what a release looks like. It targets the main branch of graph-proto-as-schema, the cosmos branch of graph-ts, and the tendermint branch of graph-as-to-rust. It doesn't manipulate any files, it just copies the output across the steps as requires and adds them to a zip file

protobuf_releases

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup GO
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Setup Node
```
Comment (kruspy) [inline .github/workflows/release.yml:21]: Shouldn't this be named Install typescript ?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          zip generated_pb_go.zip ./pb -r
+
+      - name: Setup Graph Proto as Schema
+        run: |
+          git clone https://github.com/figment-networks/graph-proto-as-schema
+          cd graph-proto-as-schema
+          yarn install
+          tsc
+          mkdir outp
+          node ./dist/index.js -o ./outp  -d ../fig/tendermint/codec/v1
```
Comment (pienkowb) [inline .github/workflows/release.yml:48]: We could pass `--typescript_namespace tendermint` here.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+      - name: Add ts files to release
+        run: |
+          zip generated_ts_files.zip ./graph-proto-as-schema/outp -r
+
+      - name: Graph-ts
+        run: |
+          git clone --branch cosmos --single-branch https://github.com/figment-networks/graph-ts
+          cp -r ./graph-proto-as-schema/outp ./graph-ts/chain
```
Comment (pienkowb) [inline .github/workflows/release.yml:57]: Have you tried running `graph-as-to-rust` on the file generated by `graph-proto-as-schema` (after putting it under `graph-ts/chain`)? Maybe we don't need the whole `graph-ts` repository.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      - name: Graph-as-to-rust
+        run: |
+          git clone --branch tendermint --single-branch https://github.com/figment-networks/graph-as-to-rust
+          cd graph-as-to-rust
+          yarn install
+          yarn tsc && node run.js
+
+      - name: Add Rust files to release
+        run: |
+          zip generated_rust_tendermint.zip ./graph-as-to-rust/tendermint.rs
```
Comment (pienkowb) [inline .github/workflows/release.yml:68]: We could rename the output file to `generated.rs` – that's how it's called in the Graph Node codebase.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "v*"
```
Comment (pienkowb) [inline .github/workflows/release.yml:6]: We could generate these files for PRs too.

File: .github/workflows/release.yml (hunk 3 above)
Comment (lavabyrd) [reply .github/workflows/release.yml]: yeah it needs some of the other files (collections etc). this is the quickest way of doing it

File: .github/workflows/release.yml (hunk 4 above)
Comment (lavabyrd) [reply .github/workflows/release.yml]: good shout

File: .github/workflows/release.yml (hunk 5 above)
Comment (lavabyrd) [reply .github/workflows/release.yml:6]: yeah i believe this will be done when we start automation PRs etc

**Comment by lavabyrd** [comment]
@pienkowb @Seb-figment @kruspy could I get you to have another look? https://github.com/lavabyrd/tendermint-protobuf-def/releases/tag/v2.1 here's an example of the generated files

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Add Protobuf files
+        run: |
+          go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.27.1
+          go mod download
+          make generate
+
+      - name: Add proto and pb files to release
+        run: |
+          zip generated_proto_files.zip ./fig -r
+          zip generated_pb_go.zip ./pb -r
```
[the remark this answers is not in the document]
Comment (lavabyrd) [reply .github/workflows/release.yml:39]: i just included them so they are all the same release page, we can ignore them if needs be, just thought they were handy for transparency

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        with:
+          go-version: 1.17
+
+      - name: Setup Typescript
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.17.1
+      - run: npm install -g typescript
+
+      - name: Setup protoc-gen-go
```
[the remark this answers is not in the document]
Comment (lavabyrd) [reply .github/workflows/release.yml:27]: ah yep. just a relic from when i was looking at protoc-gen-go. cheers!

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+      - name: Setup GO
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Setup Typescript
+        uses: actions/setup-node@v3
+        with:
+          node-version: 14.17.1
+      - run: npm install -g typescript
```
Comment (sosedoff) [inline .github/workflows/release.yml:25]: Do we need to pin this to a good version?

Comment (lavabyrd) [reply .github/workflows/release.yml:25]: yeah good shout. moved it to v16. its what they use for proto-as-rust and its building fine

**Review by pienkowb — APPROVED** [review]

**Pull request merged**